pkg/pipelines/statustracker/deployment.go

```diff
@@ -0,0 +1,153 @@
+package statustracker
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/odo/pkg/pipelines/meta"
+	"github.com/openshift/odo/pkg/pipelines/roles"
+)
+
+const (
+	operatorName   = "commit-status-tracker"
+	containerImage = "quay.io/kmcdermo/status-tracker"
+)
+
+var (
+	roleRules = []rbacv1.PolicyRule{
+		rbacv1.PolicyRule{
+			APIGroups: []string{""},
+			Resources: []string{"pods", "services", "services/finalizers", "endpoints", "persistentvolumeclaims", "events", "configmaps", "secrets"},
+			Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+		},
+		rbacv1.PolicyRule{
+			APIGroups: []string{"apps"},
+			Resources: []string{"deployments", "daemonsets", "replicasets", "statefulsets"},
+			Verbs:     []string{"create", "delete", "get", "list", "patch", "update", "watch"},
+		},
+		rbacv1.PolicyRule{
+			APIGroups: []string{"monitoring.coreos.com"},
+			Resources: []string{"servicemonitors"},
+			Verbs:     []string{"get", "create"},
+		},
+
+		rbacv1.PolicyRule{
+			APIGroups:     []string{"apps"},
+			Resources:     []string{"deployments/finalizers"},
+			ResourceNames: []string{operatorName},
+			Verbs:         []string{"update"},
+		},
+
+		rbacv1.PolicyRule{
+			APIGroups: []string{""},
+			Resources: []string{"pods"},
+			Verbs:     []string{"get"},
+		},
+
+		rbacv1.PolicyRule{
+			APIGroups: []string{"apps"},
+			Resources: []string{"replicasets", "deployments"},
+			Verbs:     []string{"get"},
+		},
+
+		rbacv1.PolicyRule{
+			APIGroups: []string{"tekton.dev"},
+			Resources: []string{"pipelineruns"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+	}
+)
```

Comment:
TypeMeta is missing. The output yaml will fail to apply because of missing "Kind".

Reply:
sure William I will look into it